hypervisor/vm.go

```diff
@@ -628,30 +640,23 @@ func (vm *Vm) Pause(pause bool) error {
 		command = "Unpause"
 	}
 
-	err := vm.GenericOperation(command, func(ctx *VmContext, result chan<- error) {
-		oldPauseState = ctx.PauseState
-		if ctx.PauseState == PauseStateBusy {
-			result <- fmt.Errorf("%s fails: earlier Pause or Unpause operation has not finished", command)
-			return
-		} else if ctx.PauseState == pauseState {
-			result <- nil
-			return
-		}
+	var err error
+	oldPauseState = ctx.PauseState
+	if ctx.PauseState == PauseStateBusy {
```

Comment:
the old code is in statemachine, so we have PauseStateBusy for state-checking and synchronization. could you add ctx.pauseLock to protect the Pause(), and remove PauseStateBusy. (also move the changes of Pause() out from this PR, create a new one with my suggestion)